packages/test/test-end-to-end-tests/src/test/benchmark/DocumentMultipleDds.ts

```diff
@@ -0,0 +1,241 @@
+				const dataStore = await this.dataObjectFactory.createInstance(
+					this.containerRuntime,
+				);
+				this.mainDataStore._root.set(`dataStore${j}`, dataStore.handle);
+			}
+			await this.waitForContainerSave(this._mainContainer);
+		}
+	}
+
+	private async waitForContainerSave(c: IContainer) {
+		if (!c.isDirty) {
+			return;
+		}
+		await new Promise<void>((resolve) => c.on("saved", () => resolve()));
+	}
+
+	/**
+	 * Creates a new DocumentCreator using configuration parameters.
+	 * @param props - Properties for initializing the Document Creator.
+	 * @param numberOfKeysInMap - Size of the document to be created 1=5Mb, 2=10Mb, etc.
+	 */
+	public constructor(private readonly props: IDocumentProps) {
+		this._dataObjectFactory = new DataObjectFactory(
+			"TestDataObject",
+			TestDataObject,
+			[SharedMap.getFactory(), SharedString.getFactory()],
+			[],
+		);
+		this.runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+			this.dataObjectFactory,
+			[[this.dataObjectFactory.type, Promise.resolve(this.dataObjectFactory)]],
+			undefined,
+			undefined,
+			runtimeOptions,
+		);
+
+		switch (this.props.documentType) {
+			case "MediumDocumentMultipleDDSs":
+				this.dsCounts = 250;
+				this.dsCountPerIteration = 125;
+				break;
+			case "LargeDocumentMultipleDDSs":
+				this.dsCounts = 500;
+				this.dsCountPerIteration = 250;
+				break;
+			default:
+				throw new Error("Invalid document type");
+		}
+	}
+
+	public async initializeDocument(): Promise<void> {
+		this._mainContainer = await this.props.provider.createContainer(this.runtimeFactory);
+		this.props.provider.updateDocumentId(this._mainContainer.resolvedUrl);
+		this.mainDataStore = await requestFluidObject<TestDataObject>(this._mainContainer, "/");
+		this.containerRuntime = this.mainDataStore._context.containerRuntime as ContainerRuntime;
+		this.mainDataStore._root.set("mode", "write");
+		await this.ensureContainerConnectedWriteMode(this._mainContainer);
+		await this.createDataStores();
+	}
+
+	/**
+	 * The loadDocument in this particular scenario does not need to do anything
+	 * as the goal is to simply measure the summarization data.
+	 * @returns the main container.
+	 */
+	public async loadDocument(): Promise<IContainer> {
+		const loader = createLoader(
+			[[this.props.provider.defaultCodeDetails, this.runtimeFactory]],
+			this.props.provider.documentServiceFactory,
+			this.props.provider.urlResolver,
+			this.props.logger,
+		);
+		return createAndAttachContainer(
+			this.props.provider.defaultCodeDetails,
+			loader,
+			this.props.provider.driver.createCreateNewRequest(this.props.provider.documentId),
+		);
+	}
+
+	private async waitForSummary(summarizer: ISummarizer): Promise<string> {
+		// Wait for all pending ops to be processed by all clients.
+		await this.props.provider.ensureSynchronized();
+		const summaryResult = await summarizeNow(summarizer);
+		return summaryResult.summaryVersion;
+	}
+
+	public async summarize(
```

Comment:
What exactly is benchmarked for these documents? I mean, which parts are run for memory / time benchmark - just summarize?

Reply:
Both load and summarize are measured for Memory and Time.   LoadDocumentAll.spec.ts and summarizationDocumentAll.spec.ts